### ISSUE TITLE
More sophisticated led patterns in led-sw

### ISSUE THREAD

**Issue by MerlijnWajer** [maemo-leste/bugtracker#491]
To be clear, low priority, but I wanted to document this at least.

Currently led-sw allows for r/g/b + on timeout and off timeout.

I wonder if we can add something for multiple points in time, and define how to get to those points.

I guess each point would be have:

* r/g/b value
* type: constant, or ramp-to-next-point
* time for point (for constant, just keep it at this value, for ramp-to-next-point, interpolate with  values of next point over time)

**Comment by MerlijnWajer** [comment]
These are the Fremantle default LED patterns:

![image](https://user-images.githubusercontent.com/517897/102713192-907cfc00-42c6-11eb-91e8-1db230d76222.png)
![image](https://user-images.githubusercontent.com/517897/102713196-94108300-42c6-11eb-808d-fb76c5f85fb7.png)
![image](https://user-images.githubusercontent.com/517897/102713197-9672dd00-42c6-11eb-9122-7a95febf4175.png)
![image](https://user-images.githubusercontent.com/517897/102713201-983ca080-42c6-11eb-865a-d1f7efcc15ba.png)

**Comment by MerlijnWajer** [comment]
Of course, these patterns would have to be written keeping in mind that changing the LEDs all the time causes a lot of wakeups, so going super fancy does not make a lot of sense, but some patterns (like: charging) could potentially be more fancy. And then users could decide if they wanted different/more fancy patterns at the cost of some battery life.

**Comment by IMbackK** [comment]
i dont agree with making led-sw more fancy at all. besides the powerconsuption concerns, every device (except pp idk there) has the ability to do patterns in hw. we should do fancy patterns there and led-sw is just a (simple) fallback
